module/documents/combatant.mjs

```diff
@@ -1,7 +1,58 @@
+import ActorDeltasField from "../data/chat-message/fields/deltas-field.mjs";
+
+/**
+ * @typedef {import("../data/chat-message/fields/deltas-field.mjs").ActorDeltasData} ActorDeltasData
+ */
+
 /**
  * Custom combatant with custom initiative roll handling.
  */
 export default class Combatant5e extends Combatant {
+  /**
+   * Create a chat message representing actor changes and displaying possible actions for this turn.
+   * @param {object} [data={}]
+   * @param {ActorDeltasData} [data.deltas]
+   * @param {string[]} [data.periods]
+   * @returns {ChatMessage5e|void}
+   */
+  async createTurnMessage({ deltas, periods }={}) {
+    const messageConfig = {
+      create: false,
+      data: {
+        speaker: ChatMessage.getSpeaker({ actor: this.actor, token: this.token }),
+        system: {
+          deltas, periods,
+          origin: {
+            combat: this.combat.id,
+            combatant: this.id
+          }
+        },
+        type: "turn",
+        whisper: game.users.filter(u => this.actor.testUserPermission(game.user, "OWNER"))
+      }
+    };
+
+    if ( !foundry.utils.isEmpty(messageConfig.data.system.deltas?.actor)
+      || !foundry.utils.isEmpty(messageConfig.data.system.deltas?.item) ) messageConfig.create = true;
+    // TODO: Also create message if actor has items with relevant activation type
+    // when implementing https://github.com/foundryvtt/dnd5e/issues/4861
+
+    /**
+     * A hook event that fires before a combat state change chat message is created.
+     * @function dnd5e.preCreateCombatMessage
+     * @memberof hookEvents
+     * @param {Combatant5e} combatant         Combatant for which the message will be created.
+     * @param {object} messageConfig
+     * @param {boolean} messageConfig.create  Should the chat message be posted?
+     * @param {object} messageConfig.data     Data for the created chat message.
+     */
+    Hooks.callAll("dnd5e.preCreateCombatMessage", this, messageConfig);
+
+    if ( messageConfig.create ) return ChatMessage.implementation.create(messageConfig.data);
```

Comment:
This looks a little unconventional with the use of `callAll` and `messageConfig.create` vs. `call` with a return of `false` cancelling the `preCreate`.

Reply:
The reason I took this approach is because there are situations where the chat message might normally not be created (when there isn't any delta data), but a hook might want to create it anyway. So if there isn't delta data the hook will be passed `{ create: false }` and then can choose to set it to `true`. I could use `Hooks.call` and provide that as another method of canceling.

Reply:
No objections from me; just wanted to note that it looked a little strange in case it wasn't intentional.